analyzers/src/SonarAnalyzer.CSharp/Rules/MesageTemplates/MessageTemplateAnalyzer.cs

```diff
@@ -0,0 +1,109 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Rules.MesageTemplates;
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class MessageTemplateAnalyzer : SonarDiagnosticAnalyzer
+{
+    private static readonly HashSet<string> MicrosoftExtensionsLoggingMethods =
```

Comment:
I've extracted https://github.com/SonarSource/sonar-dotnet/pull/8802, maybe it can be used also here.

Reply:
It probably can, but I think we can delay it since there will be two more rules using this, so we can do it in one of the next two PRs.

Reply:
Do you expect a big change? To use the method names it should be enough to delete the declarations from this class and use the other ones.